tools/ui-components/src/form-group/form-group.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import { FormGroupProps } from './types';
+
+const FormGroup = React.forwardRef<HTMLLabelElement, FormGroupProps>(
+  ({ children, className, validationState, controlId }): JSX.Element => {
+    const defaultClasses = 'mb-3.5';
+    // ToDo: validationState should change when more Form components added.
```

Comment:
I found this, while translating i will give it deeper look tomorrow https://reactjs.org/docs/typechecking-with-proptypes.html

Reply:
wouldn't passing a class  conditionally work? for instance if the validation prop is different, add a certain class to the class list.

See how classes change based on prop for your reference.

Reply:
https://github.com/freeCodeCamp/freeCodeCamp/blob/main/tools/ui-components/src/alert/alert.tsx

Reply:
Here is my lack of experience with React is obvious,

I am worried about passing class conditionally like we are doing it in `Alert`, because the `Alert` component is re-rendered when the `message` is `error` or alike

https://github.com/freeCodeCamp/freeCodeCamp/blob/10d626dd235d25d56cdb8d70893fb707e9b040ae/client/src/components/formHelpers/form-fields.tsx#L60-L73

Here is what I am worried about anyway.

1- Re rendering the `FormGroup` component when the `input` is invalid, will delete the input that the users typed.
2- It will hinder performance, because it's creating a new component every time the input is invalid and the `FormGroup` requires a new class.

---- 

what I want `FormGroup` and `FormControl` to not be re-rendered, and receive a certain class based on input state.

Reply:
- The input is controlled by the parent of the form group in a state so the value will be passed to form group's children.
- I would not worry about that since we have similar practices in our codebase.